lib/private/Authentication/Token/PublicKeyTokenProvider.php

```diff
@@ -435,11 +446,12 @@ public function updatePasswords(string $uid, string $password) {
 
 		// Update the password for all tokens
 		$tokens = $this->mapper->getTokenByUser($uid);
+		$passwordHash = $this->hashPassword($password);
 		foreach ($tokens as $t) {
 			$publicKey = $t->getPublicKey();
-			$encryptedPassword = $this->encryptPassword($password, $publicKey);
-			if ($t->getPassword() !== $encryptedPassword) {
-				$t->setPassword($encryptedPassword);
+			if ($t->getPasswordHash() === null || $this->hasher->verify(sha1($password) . $password, $t->getPasswordHash())) {
```

Comment:
Did you maybe mean:
```suggestion
			if ($t->getPasswordHash() === null || !$this->hasher->verify(sha1($password) . $password, $t->getPasswordHash())) {
```

So the password is updated everytime it does NOT match the old stored password?

Reply:
But the actual problem is `$this->hasher->verify` takes like ~0.2 seconds